src/dreamchecker/lib.rs

```diff
@@ -1043,17 +1043,37 @@ impl<'o, 's> AnalyzeProc<'o, 's> {
         }
     }
 
-    fn visit_unary(&mut self, rhs: Analysis<'o>, op: &UnaryOp) -> Analysis<'o> {
+    // crement = increment/decrement
+    fn check_crement(&mut self, rhs: Analysis<'o>, location: Location, crement: &str) -> Analysis<'o> {
+        let typeerror;
+        match rhs.static_ty {
+            StaticType::None => {
+                return Analysis::empty()
+            },
+            StaticType::Type(typeref) => {
+                if let Some(proc) = typeref.get_proc(crement) {
+                    return Analysis::empty()
```

Comment:
It would be nice if this called `visit_proc` for consistency

Reply:
oh good point!

Reply:
- Where is the `null` argument coming from?
- What does `x++` evaluate to if `operator++` is overloaded? Is `operator++`'s return value really thrown away?
- `exact` should probably be `false` because a subtype's ++ might be called.

You can fix the compile error with `let args = [...];` and `..., &args, ...`.